decidim/decidim
# Add serializer customization via listener

#### :tophat: What? Why?
When exporting data, admin users want different fields to be included into export. This pull request adds notification event that publishes before we return serialized data. Because this notification event, it's possible to add subscriber to a config file, which listens this serialize event and can modify serialize data.

#### :pushpin: Related Issues
https://meta.decidim.org/processes/roadmap/f/122/proposals/14588

#### Testing
1. Add initializer where you subscribe to serialize event, for example:
```
initializer "decidim_budgets.serializer_listener" do
  ActiveSupport::Notifications.subscribe("decidim.serialize.budgets.project_serializer") do |_event_name, data|
    data[:serialized_data][:test_field] = "Resource class: #{data[:resource].class}"
  end
end
```
2. Restart development app
3. Go to admin panel and export projects
4. Check your mail 

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.adoc) to this repository.


:hearts: Thank you!
